### PR TITLE
encryption-and-compression-functions: Update for MySQL 8.0

### LINKED ISSUE #16819: Update reference manual links in functions-and-operators/encryption-and-compression-functions.md

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

Closes #16819 

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/encryption-and-compression-functions.md [hunk 1]
```diff
@@ -25,11 +25,18 @@ TiDB supports most of the [encryption and compression functions](https://dev.mys
 | [`UNCOMPRESSED_LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/encryption-functions.html#function_uncompressed-length)                             | Return the length of a string before compression  |
 | [`VALIDATE_PASSWORD_STRENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/encryption-functions.html#function_validate-password-strength) | Validate the password strength |
 
+
```
Comment (mjonss) [inline functions-and-operators/encryption-and-compression-functions.md:28]: Is this needed?

File: functions-and-operators/encryption-and-compression-functions.md [hunk 2]
```diff
@@ -25,11 +25,18 @@ TiDB supports most of the [encryption and compression functions](https://dev.mys
...
+The [`validate_password.*`](/system-variables.md) variables influence the `VALIDATE_PASSWORD_STRENGTH()` function.
 
 ## Unsupported functions
 
 * `DES_DECRYPT()`, `DES_ENCRYPT()`, `OLD_PASSWORD()`, `ENCRYPT()`: these functions were deprecated in MySQL 5.7 and removed in 8.0.
 * Functions only available in MySQL Enterprise [Issue #2632](https://github.com/pingcap/tidb/issues/2632).
+
+## MySQL Compatibility
+
+* TiDB doesn't support the `STATEMENT_DIGEST()` and `STATEMENT_DIGEST_TEXT()` functions.
```
Comment (mjonss) [inline functions-and-operators/encryption-and-compression-functions.md:42]: Should we create an issue in tidb repository for adding these?

**Review by mjonss — APPROVED** [review]
LGTM, with some minor questions.

File: functions-and-operators/encryption-and-compression-functions.md (hunk 1 above)
Comment (dveeden) [reply functions-and-operators/encryption-and-compression-functions.md:28]:
```suggestion
```

File: functions-and-operators/encryption-and-compression-functions.md (hunk 2 above)
Comment (dveeden) [reply functions-and-operators/encryption-and-compression-functions.md:41]: I don't think that's needed. But I'm also not against it...

**Review by bb7133 — APPROVED** [review]
LGTM

File: functions-and-operators/encryption-and-compression-functions.md [hunk 3]
```diff
@@ -27,9 +27,15 @@ TiDB supports most of the [encryption and compression functions](https://dev.mys
 
 ## Related system variables
 
-The `block_encryption_mode` variable sets the encryption mode that is used for `AES_ENCRYPT()` and `AES_DECRYPT()`.
+The [`block_encryption_mode`](/system-variables.md#block_encryption_mode) variable sets the encryption mode that is used for `AES_ENCRYPT()` and `AES_DECRYPT()`.
+
+The [`validate_password.*`](/system-variables.md) variables influence the `VALIDATE_PASSWORD_STRENGTH()` function.
```
Comment (qiancai) [inline functions-and-operators/encryption-and-compression-functions.md:32]:
```suggestion
The [`validate_password.*`](/system-variables.md) variables affect the `VALIDATE_PASSWORD_STRENGTH()` function.
```

File: functions-and-operators/encryption-and-compression-functions.md [hunk 4]
```diff
@@ -27,9 +27,15 @@ TiDB supports most of the [encryption and compression functions](https://dev.mys
...
 
 ## Unsupported functions
 
 * `DES_DECRYPT()`, `DES_ENCRYPT()`, `OLD_PASSWORD()`, `ENCRYPT()`: these functions were deprecated in MySQL 5.7 and removed in 8.0.
 * Functions only available in MySQL Enterprise [Issue #2632](https://github.com/pingcap/tidb/issues/2632).
+
+## MySQL Compatibility
+
+* TiDB doesn't support the `STATEMENT_DIGEST()` and `STATEMENT_DIGEST_TEXT()` functions.
```
Comment (qiancai) [inline functions-and-operators/encryption-and-compression-functions.md:41]:
```suggestion
## MySQL compatibility

* TiDB does not support the `STATEMENT_DIGEST()` and `STATEMENT_DIGEST_TEXT()` functions.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**